pkg/antctl/raw/supportbundle/command.go

```diff
@@ -245,19 +261,61 @@ func download(suffix, downloadPath string, client *rest.RESTClient, component st
 	return nil
 }
 
-func downloadAll(agentClients map[string]*rest.RESTClient, controllerClient *rest.RESTClient, downloadPath string, bar *pb.ProgressBar) error {
-	return mapClients(
+func writeFailedNodes(downloadPath string, nodes []string) error {
+	file, err := os.OpenFile(downloadPath+"/failed_nodes", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("err create file for failed nodes: %w", err)
+	}
+	dataWriter := bufio.NewWriter(file)
+	for _, node := range nodes {
+		_, _ = dataWriter.WriteString(node + "\n")
+	}
+	err = dataWriter.Flush()
+	if err != nil {
+		return err
+	}
+	err = file.Close()
```

Comment:
add defer clause to file.Close()

Reply:
Sorry, rocky mistake. Updated. Will be more carefully next time.